tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py

```diff
@@ -0,0 +1,256 @@
+
+from scipy import interpolate  # NOQA
+
+
+@testing.with_requires("scipy")
+class TestBarycentric:
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-6, rtol=1e-6)
+    def test_lagrange(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(test_xs)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-5, rtol=1e-5)
+    def test_scalar_1(self, xp, scp):
+        true_poly = xp.poly1d([-1, 2, 6, -3, 2])
+        xs = xp.linspace(-1, 1, 10)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(xp.array(7))
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-5, rtol=1e-5)
+    def test_scalar_2(self, xp, scp):
+        true_poly = xp.poly1d([-1, 2, 6, -3, 2])
+        xs = xp.linspace(-1, 1, 10)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(7)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_delayed(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs)
+        P.set_yi(ys)
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_append(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs[:3], ys[:3])
+        P.add_xi(xs[3:], ys[3:])
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-5, rtol=1e-5)
+    def test_vector(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        xs = xp.array([0, 1, 2], dtype=dtype)
+        ys = xp.array([[0, 1], [1, 0], [2, 1]], dtype=dtype)
+        test_xs = xp.linspace(-1, 3, 100, dtype=dtype)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_scalarvalue_1(self, xp, scp, dtype):
```

Comment:
We may use `pytest.mark.parametrize` to test some inputs in a method. Lambda expressions can be used to make cupy/numpy ndarray input lazily. (Same for test_shapes_vectorvalue and test_shapes_1d_vectorvalue)

```py
@pytest.mark.parametrize('input', [0, lambda xp, dtype: xp.array(0, dtype=dtype), [0], [0, 1])
def test_shapes_scalarvalue(self, xp, scp, dtype, input):
    ...
    if callable(input):
        input = input(xp, dtype)
    return xp.shape(P(input))
```

Reply:
ah, relating to https://github.com/cupy/cupy/pull/6958/files#r939783252 for the inputs

Reply:
Using lambda and list comprehensions sounds great to me. Thanks! :)

I've updated it.